tools/ExamplesGenerator.ps1

```diff
@@ -0,0 +1,268 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+Param(
+    $ModulesToGenerate = @(),
+    [string] $ModuleMappingConfigPath = (Join-Path $PSScriptRoot "..\config\ModulesMapping.jsonc")
+)
+function Get-GraphMapping {
+    $graphMapping = @{}
+    $graphMapping.Add("v1.0", "examples\v1.0")
+    $graphMapping.Add("beta", "examples\v1.0-beta")
+    return $graphMapping
+}
+
+function Start-Generator {
+    Param(
+        $ModulesToGenerate = @()
+    )
+
+    $GraphMapping = Get-GraphMapping 
+    $GraphMapping.Keys | ForEach-Object {
+        $graphProfile = $_
+        Get-FilesByProfile -GraphProfile $graphProfile -GraphProfilePath $GraphMapping[$graphProfile] -ModulesToGenerate $ModulesToGenerate 
+    }
+}
+function Get-FilesByProfile {
+    Param(
+        [ValidateSet("beta", "v1.0")]
+        [string] $GraphProfile = "v1.0",
+        [ValidateNotNullOrEmpty()]
+        [string] $GraphProfilePath = "examples\v1.0",
+        [ValidateNotNullOrEmpty()]
+        $ModulesToGenerate = @()
+    )
+
+
+    $ModulesToGenerate | ForEach-Object {
+        $ModuleName = $_
+        $modulePath = Join-Path $PSScriptRoot "..\src\$ModuleName\$ModuleName\$GraphProfilePath"
+        $OpenApiFile = Join-Path $PSScriptRoot "..\openApiDocs\v1.0\$ModuleName.yml"
+        #test this path first before proceeding
+        if ((Test-Path $OpenApiFile)) {
+        $yamlContent = Get-Content -Path $OpenApiFile
+        $OpenApiContent = ($yamlContent | ConvertFrom-Yaml)
+        Get-Files -GraphProfile $GraphProfile -GraphProfilePath $modulePath -Module $ModuleName -OpenApiContent $OpenApiContent
+        }
+    }
+
+}
+function Get-Files {
+    param(
+        [ValidateSet("beta", "v1.0")]
+        [string] $GraphProfile = "v1.0",
+        [ValidateNotNullOrEmpty()]
+        [string] $GraphProfilePath = (Join-Path $PSScriptRoot "..\src\Users\Users\examples\v1.0"),
+        [ValidateNotNullOrEmpty()]
+        [string] $Module = "Users",
+        [Hashtable] $OpenApiContent 
+    )
+    $ModuleManifestFile = (Join-Path $PSScriptRoot "..\src\$Module\$Module\Microsoft.Graph.$Module.psd1")
+    $ModuleManifestFileContent = Get-Content -Path $ModuleManifestFile
+
+    try {
+        if ((Test-Path $GraphProfilePath)) {
+            
```

Comment:
Please address the formatting issues in the script (extra parenthesis, unnecessary new lines, etc).
```suggestion
        if (Test-Path $GraphProfilePath) {

```